### PR TITLE
Transformer improvements

### PR DESCRIPTION
**Pull request by ApplyBoardDak**

### CODE REVIEW

File: stories/components/slate-editor.tsx [hunk 1]
```diff
@@ -161,6 +161,9 @@ const renderLeaf = ({ attributes, children, leaf }: RenderLeafProps) => {
   if (leaf.inlineCode) {
     children = <code>{children}</code>;
   }
+  if (leaf.directive === 'notice') {
+    children = <span style={{ backgroundColor: 'yellow',  color: leaf.color }}>{children}</span>
```
Comment (BenLorantfy-AB) [inline stories/components/slate-editor.tsx:165]: shouldn't we use `intent` here?  Not `color` ?

File: fixtures/article.md [hunk 2]
```diff
@@ -93,3 +93,5 @@ function $initHighlight(block, cls) {
 
 export  $initHighlight;
 ```
+
+:notice[This **content** is a generic directive!]{color=blue}
```
Comment (BenLorantfy-AB) [inline fixtures/article.md:97]: Is this just an example?  This PR adds support for any directive and any attributes?

File: package.json
Comment (BenLorantfy-AB) [inline package.json:1]: I wonder if this PR is something we could contribute upstream?  Maybe we should ask Sasha if we can do this?  That way we don't need to maintain this fork

**Review by BenLorantfy-AB — APPROVED** [review]
I'm not super familiar with this library but generally looks good 👍

File: fixtures/article.md (hunk 2 above)
Comment (ApplyBoardDak) [reply fixtures/article.md:97]: yeah, this is just an example

File: package.json
Comment (ApplyBoardDak) [reply package.json:1]: 'twould be nice but our fork is an older version of the library because of ESM shenanigans. add it to the to-do list 😎

**Pull request merged**